src/Providers/CustomerProvider.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * CustomerProvider class file.
+ *
+ * @package Automattic\WooCommerce\Providers
+ */
+
+namespace Automattic\WooCommerce\Providers;
+
+use \WC_Customer;
+
+/**
+ * Provides methods to retrieve customer objects.
+ */
+class CustomerProvider {
```

Comment:
To me this feels more like a `CustomerRepository` than a `CustomerProvider`. I'm also thinking we might want to wait to create things like this until after we've taken a look at improvements we might make in the data access layer and models themselves. If we create a public API for fetching customers in this, but want to change the return type to a different model later, we won't be able to.

Reply:
The `new WC_Customer( get_current_user_id(), true )` is a pattern I see used often and I thought it was worth encapsulating it in a method with a proper name, it's a "helper" method more than a repository-ish thing.